src/main/java/org/embulk/output/GcsOutputPlugin.java

```diff
@@ -78,6 +88,26 @@ public TransactionalFileOutput open(TaskSource taskSource, final int taskIndex)
         return new GcsTransactionalFileOutput(task, client, taskIndex);
     }
 
+    private void deleteFiles(PluginTask task)
+    {
+        logger.info("delete files");
+        Storage client = createClient(task);
+        try {
+            Objects listResult = client.objects().list(task.getBucket()).setDelimiter("/").setPrefix(task.getPathPrefix()).execute();
+            if (listResult.getItems() == null) {
+                logger.info("no files found");
+                return;
+            }
+            for (StorageObject item : listResult.getItems()) {
+                logger.info("delete file: {}/{}", item.getBucket(), item.getName());
+                client.objects().delete(item.getBucket(), item.getName()).execute();
```

Comment:
Better to move `logger.info("delete file: {}/{}", item.getBucket(), item.getName());` after sending requests.
Only logging message will be shown in case delete operation fails,

Reply:
Could you also use retry with exponential backoff?